instrumentation/trace-annotation/src/test/groovy/TraceProvidersTest.groovy

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import io.opentelemetry.auto.test.AgentTestRunner
+import io.opentelemetry.auto.test.utils.ConfigUtils
+import io.opentelemetry.test.annotation.SayTracedHello
+
+/**
+ * This test verifies that Otel supports various 3rd-party trace annotations
+ */
+class TraceProvidersTest extends AgentTestRunner {
+
+  static {
+    ConfigUtils.updateConfig {
+      System.clearProperty("ota.trace.annotations")
+      System.setProperty("ota.trace.classes.exclude", TraceProvidersTest.name + "*")
+    }
+  }
+
+  def specCleanup() {
+    ConfigUtils.updateConfig {
+      System.clearProperty("ota.trace.classes.exclude")
+    }
+  }
+
```

Comment:
```suggestion
```
Is this needed?

Reply:
Specifying `ota.trace.classes.exclude` will stop instrumenting this test class itself. For me it polluted the log during test run. And I assume it adds some overhead to test running as well. I think all tests would benefit from this exclusion.
Having non-empty `ota.trace.annotations` will turn off default annotations list and this will result in this test failing.

Thus I think this configuration should stay.

Reply:
Ah, makes sense. Can that change be made globally in `AgentTestRunner` (with a comment, so we don't accidentally remove it later thinking it's not needed)? Also, maybe this change would be better in a separate PR?

Reply:
So I will leave this PR as it is and will do a new PR where I will add `ota.trace.classes.exclude` to `AgentTestRunner`? Sounds good.

Reply:
👍

Reply:
@trask Unfortunately it turns out that we cannot just globally apply class exclusion for all test classes. Some of them, e.g. `ExecutorInstrumentationTest` expect that their inner classes get instrumented during test. So that would require reviewing existing test and stricter rules for future ones. Will drop this until the benefit is clear.

Reply:
ok, makes sense. do u mind removing, or adding a comment explaining, the usage in `TraceProvidersTest.groovy` then?